src/properties/property-effects.html

```diff
@@ -190,26 +176,28 @@
    */
   function runNotifyEffects(inst, props, oldProps) {
     // Notify
-    let notified;
-    let notifyEffects = inst[TYPES.NOTIFY];
-    let id = effectUid++;
-    // Try normal notify effects; if none, fall back to try path notification
-    // TODO(kschaaf) This is a hot path which we could avoid if (1) no
-    // "notify" effects AND (2) if we knew there were no paths to notify. This
-    // 2nd piece of info is not currently available but could be added perhaps
-    // in `_setProperty`.
-    for (let prop in props) {
-      if (notifyEffects && runEffectsForProperty(inst, notifyEffects, id,
-        prop, props[prop], oldProps && oldProps[prop])) {
-        notified = true;
-      } else if (notifyPath(inst, prop, props[prop])) {
-        notified = true;
+    let notifyEffects = inst.__notifyEffects;
+    if (notifyEffects) {
```

Comment:
To investigate: seems like we shouldn't be able to get away with this perf optimization because the 1.x contract was that path notifications always fire (regardless of existence of a notify effect for the root property, which is the whole reason notify effects are special cased in this function).  I put this in provisionally while I was looking at separating path & property code, but need to look into why this isn't causing tests to fail.  I think `if (notifyEffects || this.__dataHasPathsCounter)` is probably enough to avoid the `for` loop over props if there are no notify effects.